include/seqan/index/index_fm_rank_dictionary_levels.h

```diff
@@ -345,35 +384,50 @@ struct RankDictionaryEntry_<TValue, Levels<TSpec, TConfig> >
 // Metafunction RankDictionaryBitMask_
 // ----------------------------------------------------------------------------
 
-// TODO(cpockrandt): simplify
 template <typename TWord>
-constexpr TWord _bitmask2(unsigned const bitsTotal, unsigned blocks, unsigned const constBlocks, unsigned const blocksize, TWord const value)
+constexpr TWord _bitmaskRecursive(unsigned const bitsUsed, unsigned const blocksLeft, unsigned const blocksProcessed, unsigned const blocksize, TWord const value1, TWord const value2, TWord const result)
 {
-    return (blocks == constBlocks) ?
-           _bitmask2<TWord>(bitsTotal, blocks - 1, constBlocks, blocksize, value << (bitsTotal - blocksize)) :
-           (
-               (blocks == 0) ?
-               value >> (bitsTotal % blocksize) :
-               _bitmask2<TWord>(bitsTotal, blocks - 1, constBlocks, blocksize, value | (value >> blocksize))
-           );
+    return (blocksLeft == 0) ?
+           result :
+           _bitmaskRecursive<TWord>(bitsUsed, blocksLeft - 1, blocksProcessed + 1, blocksize, value1, value2, result | (((blocksProcessed % 2) ? value1 : value2) << (bitsUsed - blocksProcessed * blocksize)));
```

Comment:
I usually use this syntax for the ternary operator:
```c++
return (a < b)
    ? true
    : false;
```
Also please make sure that total line length doesn't exceed 120characters; move the return type into an extra line and put each parameter in its own line.